js/apps/system/_admin/aardvark/APP/react/public/assets/src/mode-aql.js

```diff
@@ -132,34 +139,67 @@ var AqlHighlightRules = function() {
             next : "comment_ml"
         }, {
             token : "string",           // " string
-            regex : '".*?"'
+            regex : '"',
+            next: "string_double"
         }, {
             token : "string",           // ' string
-            regex : "'.*?'"
+            regex : "'",
+            next: "string_single"
+        }, {
+            token : "identifier",           // ` quoted identifier
+            regex : "`",
+            next: "identifier_backtick"
         }, {
-            token : "string",           // ` string
-            regex : "`.*?`"
+            token : "identifier",           // ´ quoted identifier
+            regex : "´",
+            next: "identifier_forwardtick"
         }, {
             token : "constant.numeric", // binary integer
-            regex : "0[bB][01]+\\b"
+            regex : /0[bB][01]+\b/
         }, {
             token : "constant.numeric", // hexadecimal integer
-            regex : "0[xX][0-9a-fA-F]+\\b"
+            regex : /0[xX][0-9a-fA-F]+\b/
         }, {
             token : "constant.numeric", // float
-            regex : "[+-]?\\d+(?:(?:\\.\\d*)?(?:[eE][+-]?\\d+)?)?\\b"
+            regex : /(?:(?:0|[1-9][0-9]*)(?:\.[0-9]+)?|\.[0-9]+)(?:[eE][\-\+]?[0-9]+)?/,
         }, {
-            token : keywordMapper,
-            regex : "[a-zA-Z_][a-zA-Z0-9_]*\\b"
+            token : "constant.numeric", // decimal integer
+            regex : /0|[1-9][0-9]*\b/,
+        }, {
+            token : "variable.global",
+            regex : "@" + aqlBindVariablePattern
+        }, {
+            token : "variable",
+            regex : aqlBindVariablePattern
         }, {
             token : "keyword.operator",
-            regex : "\\+|\\-|\\/|\\/\\/|%|@>|<@|&&|\\|\\||!|<|>|<=|=>|==|!=|=|\\[\\*\\]"
+            regex : /=~|!~|==|!=|>=|>|<=|<|=|!|&&|\|\||\+|\-|\*|\/|%|\?|::|:|\.\./,
         }, {
             token : "paren.lparen",
-            regex : "[\\(\\{]"
+            regex : /[\(\{\[]/
         }, {
             token : "paren.rparen",
-            regex : "[\\)\\}]"
+            regex : /[\)\}\]]/
+        }, {
+            token : "punctuation",
+            regex : /[\.,]/
+        }, {
+            token : "keyword",  // language construct usable in COLLECT
+            regex : /WITH\s+COUNT\s+INTO\b/,
+            caseInsensitive: true
+        }, {
+            token : "keyword",  // handle separately because of the whitespace
+            regex : /AT\s+LEAST\b/,
+            caseInsensitive: true // BUG: https://github.com/ajaxorg/ace/issues/4887
+        }, {
+            token : "language.variable", // case sensitive
+            regex : /(?:CURRENT|NEW|OLD)\b/
+        }, {
+            token : keywordMapper,
+            regex : /[a-zA-Z_][a-zA-Z0-9_]*\b/
+        }, {
+            token : "identifier",
+            regex : /(?:\$?|_+)[a-zA-Z]+[_a-zA-Z0-9]*/,
```

Comment:
Could use this regex for the keywordMapper instead, which makes all non-matching tokens "identifier" anyway.